### PR TITLE
[Documentation] 03-Database infrastructure (rebased to loris-mri-docs)

### PR DESCRIPTION
**Pull request by cmadjar**
This is a first draft of section 4 about the backend directory structure and the database infrastructure.

### CODE REVIEW

File: docs/04-TechnicalInfrastructure.md [hunk 1]
```diff
@@ -1 +1,323 @@
...
+    `CandID/Visit_label`. Whithin each of these visit labels, data are first 
+    organized by imaging type (`mri` or `pet` for example) and then by output 
+    type (such as `native` or `processed`). For example, a native T1W image for 
+    subject 123456's V1 visit will be located in 
+    `data/assembly/123456/V1/mri/native/Project_123456_V1_T1W_001.mnc`.
+    
+```
+## Content of the /data/project/data/assembly directory
+.
+|__ candid
```
[the remark this answers is not in the document]
Comment (davidblader) [reply docs/04-TechnicalInfrastructure.md]: applies to following directory diagrams as well

File: docs/04-TechnicalInfrastructure.md [hunk 2]
```diff
@@ -1 +1,323 @@
...
+        |__ Project_DTIPrep_XML_protocol.xml
+```
+
+## 4.2 Database infrastructure
+
+The database infrastructure is divided in six main components based on the 
+  workflow happening from native images insertion to quality control and 
+  ultimately insertion of processed datasets.  
+
+![overall_DB_structure](images/overall_DB_structure.png)
```
Comment (davidblader) [inline docs/04-TechnicalInfrastructure.md]: diagrams are really good! there is just one small mistake for the type of CandID (`int(10)`), which appears to have taken on the datatype of CenterID (`tinyint(2)`)
i'm not sure how you made these diagrams, so if it is too much trouble then i would not worry about it

File: docs/04-TechnicalInfrastructure.md [hunk 3]
```diff
@@ -1 +1,323 @@
...
+    |   |__ DTI_QC`date`.log
+    |__ DTIPrep_register
+    |   |__ DTIregister`date`.log
+    |   |__ DTIregister`date`.log
+    |__ registerProcessed
+        |__ registerProcessed`date`.log
+        |__ registerProcessed`date`.log
+```
+
+- Incoming scans from the Imaging uploader module (or automatic cron jobs) are
```
[the remark this answers is not in the document]
Comment (cmadjar) [reply docs/04-TechnicalInfrastructure.md]: Could we talk about this one tomorrow (Tuesday)? I just want to make sure I understand correctly what you are suggesting :)

File: docs/04-TechnicalInfrastructure.md [hunk 4]
```diff
@@ -1 +1,323 @@
...
+    |__file.mnc
+```
+     
+- The MINC images that can be viewed via BrainBrowser in the imaging browser 
+    module are located under the `data/assembly` directory and organized by 
+    `CandID/Visit_label`. Whithin each of these visit labels, data are first 
+    organized by imaging type (`mri` or `pet` for example) and then by output 
+    type (such as `native` or `processed`). For example, a native T1W image for 
+    subject 123456's V1 visit will be located in 
+    `data/assembly/123456/V1/mri/native/Project_123456_V1_T1W_001.mnc`.
```
[the remark this answers is not in the document]
Comment (cmadjar) [reply docs/04-TechnicalInfrastructure.md]: wow! Nice catch! :)

**Comment by cmadjar** [comment]
@davidblader, thank you for your comments!!! Very helpful!! 
I think I fixed everything that you brought up except the point that was not clear to me. I also made the violated table section clearer now that we have more answers.

**Pull request merged**